### PR TITLE
docs(google drive): add Google Drive OOTB connector doc

### LINKED ISSUE #1161: Google Drive: Add docs
Related to:
- https://github.com/camunda/product-hub/issues/237
- https://github.com/camunda/product-hub/issues/239
- https://github.com/camunda/cloud-connector-google-drive/issues/2

This task is to create an official documentation for the Google Drive Connector - Create Folder and Create File.

### PR DESCRIPTION
**Pull request by igpetrov**
closes #1161

### CODE REVIEW

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+id: googledrive
+title: Google Drive connector
+description: Create empty folders or documents from templates from your BPMN process.
+---
+
+The **Google Drive connector** allows you to create empty folders or documents from templates from your BPMN process.
+
+## Prerequisites
+
+To start working with the **Google Drive connector**, a relevant OAuth token must be configured and stored as a secret in the user's cluster. The token has to have permission to read and create a folder and/or files from a desired Google Drive instance. Follow the steps from the appendix to find out more about creating an OAuth token and giving relevant permissions.
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:11]: Let's link the `appendix&faq`

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
+---
+id: googledrive
+title: Google Drive connector
+description: Create empty folders or documents from templates from your BPMN process.
+---
+
+The **Google Drive connector** allows you to create empty folders or documents from templates from your BPMN process.
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:7]: Maybe a bit more verbose

```suggestion
The **Google Drive connector** allows you to create empty folders or documents on [Google Drive](https://drive.google.com/) from templates from your BPMN process.
```

Comment (MaxTru) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:7]: Is there a link we can add for `templates` (e.g., I have not used Google Drive templates) to get started with what they are?

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+## Prerequisites
+
+To start working with the **Google Drive connector**, a relevant OAuth token must be configured and stored as a secret in the user's cluster. The token has to have permission to read and create a folder and/or files from a desired Google Drive instance. Follow the steps from the appendix to find out more about creating an OAuth token and giving relevant permissions.
+
+## Create a Google Drive connector task
+
+Currently, the **Google Drive connector** supports two flavors: create a new folder and create a file from a template.
+
+To use a **Google Drive connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu or create a new connector task by using the **Append connector** context menu. Follow our guide on using connectors to learn more.
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:17]: > Follow our guide on using connectors to learn more.

Should be linked

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+![Google Drive connector new folder example](../img/connectors-googledrive-new-folder-filled.png)
+
+1. In the **Authentication** section, set the field **Google Drive Bearer Token** to your OAuth token, for example, `secrets.GDRIVE_OAUTH_TOKEN`.
+2. In the **Select Operation** section, set the field value **Operation Type** as **Create Folder**.
+3. In the **Operation Details** section, set the field **Folder name** as the desired name of a folder you wish to create. For example, `MyNewFolder`. Alternatively, you could use a FEEL expression.
+4. _(optional)_ In the **Operation Details** section, set the field **Parent folder ID** to the desired parent, inside which a new folder will be created. Please, keep in mind, that if not specified, a new folder will be created in the Google Drive root folder of a user, who owns the OAuth token.
+5. _(optional)_ In the **Operation Details** section, set the field **Additional properties or metadata**, set Google Drive compatible properties or metadata. This property requires FEEL input. Check the appendix for known values and limitations.
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:29]: Link to appendix

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+2. In the **Select Operation** section, set the field value **Operation Type** as **Create File from template**.
+3. In the **Operation Details** section, set the field **File name** as the desired name of a folder you wish to create. You can use FEEL expressions here.
+4. In the **Operation Details** section, set the field **Template ID** of the desired template.
+5. _(optional)_ In the **Operation Details** section, set the field **Parent folder ID** to the desired parent, inside which a new file will be created. Please, keep in mind, that if not specified, a new folder will be created in the Google Drive root folder of a user, who owns the OAuth token.
+6. In the **Operation Details** section, set the field **Template variables** as desired variables that will be applied to the template. The template variables a compatible with the Google Docs Requests API. This property requires FEEL input.
+7. _(optional)_ In the **Operation Details** section, set the field **Additional properties or metadata**, set Google Drive compatible properties or metadata. This property requires FEEL input. Check the appendix for known values and limitations.
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:43]: Link to appendix

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+## Google Drive connector response
+
+The **Google Drive connector** exposes Google Drive API response as a local variable called response.
+The following fields are available in the response variable:
+`googleDriveResourceId` - ID of a newly created resource
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:49]:
```suggestion
`googleDriveResourceId` - ID of the newly created resource
```

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+`googleDriveResourceUrl` - human-readable URL of a newly created resource
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:50]:
```suggestion
`googleDriveResourceUrl` - human-readable URL of the newly created resource
```

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+You can use an Output Mapping to map the response:
+
+1. Use **Result Variable** to store the response in a process variable. For example, `myResultVariable`.
+2. Use **Result Expression** to map fields from the response into process variables. For example:
+
+```
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:57]: I don't get the example. The expression you put would map static fields into variables right?

So after execution the variable `googleDriveResourceId` would exist in my process with the value `1KMG4S_18kLNZa7ZCARtGIh8DmIoXXXXXXXXX`. Why would I do that?

Would a more appropriate example not be something like this:

```
{
  "newFolderID": response.googleDriveResourceId
}
```

Comment (MaxTru) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:57]: Or do I miss something?

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+  "googleDriveResourceId":"1KMG4S_18kLNZa7ZCARtGIh8DmIoXXXXXXXXX",
+  "googleDriveResourceUrl":"https://docs.google.com/document/d/1KMG4S_18kLNZa7ZCARtGIh8DmIoXXXXXXXXX"
+}
+```
+
+# Appendix & FAQ
+
+## How can I get Google Drive OAuth token?
+
+Refer to the official Google Drive API documentation page.
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:68]: Link missing

**Review by MaxTru — CHANGES_REQUESTED** [review]
This connector already looks super nice!

I did a quick first review and found some things, primarily linking.

In general missing is update of `sidebars.js` (please also ensure your change works end-2-end by running `npm run start`)

On top (once we have this close to be ready for merge), we should also make the change in the `versioned_docs/version-8.0` (at least if we release this prior to October, which I assume). Otherwise the change would not be visible in the docs before the 8.1 release (in October)

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 1 above)
Comment (christinaausley) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:11]: Agreed here.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+The **Google Drive connector** allows you to create empty folders or documents from templates from your BPMN process.
+
+## Prerequisites
+
+To start working with the **Google Drive connector**, a relevant OAuth token must be configured and stored as a secret in the user's cluster. The token has to have permission to read and create a folder and/or files from a desired Google Drive instance. Follow the steps from the appendix to find out more about creating an OAuth token and giving relevant permissions.
+
+## Create a Google Drive connector task
+
+Currently, the **Google Drive connector** supports two flavors: create a new folder and create a file from a template.
```
Comment (christinaausley) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:15]: What do you mean by "flavors" here? Use cases? Instances?

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 3 above)
Comment (christinaausley) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:17]: When you say "Follow our guide on using connectors to learn more" -- please link to the connectors guide.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 4 above)
Comment (christinaausley) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:29]: "set the field **Additional properties or metadata** to..."?

I'm confused by "In the **Operation Details** section, set the field **Additional properties or metadata**, set Google Drive compatible properties or metadata" as it isn't a complete sentence. What do you mean here?

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 5 above)
Comment (christinaausley) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:43]: Same comment here.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 9 above)
Comment (christinaausley) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:68]: Please link to this 👍

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+
+![Google Drive connector get template ID](../img/connectors-googledrive-get-template-id.png)
+
+## Can you show me an example of a valid template?
+
+The valid template looks like the one on a screenshot.
+
+![Google Drive connector document template example](../img/connectors-googledrive-template-example.png)
+
+Now, in the Template variables field we can apply the following FEEL JSON object that has to be compatible with the Google Docs Requests API:
```
Comment (christinaausley) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:122]: Is "Template variables" field a UI element? If so, it should be bolded.

**Review by christinaausley — CHANGES_REQUESTED** [review]
In addition to @MaxTru 's comments on adding links to resources as appropriate, I have a few comments for clarification.

You'll also see a commit labeled `style(formatting): technical review` come through where I've adjusted some of the grammar, syntax, and punctuation according to our style guide. We can't have H1 headers in our docs, so you'll see I've adjusted this as well. Please let me know if you have any questions about this. Thanks!

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 10 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:15]: Two variants of templates (and use cases) - one to create a folder, one to create a file.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 4 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:29]: What is meant here, there's a section called **Operation Details**. This section contains an input field **Additional properties or metadata**. In this section, user can input Google Drive compatible properties, e.g. _description_ of the folder.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 10 above)
Comment (christinaausley) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:15]: Excellent! I would phrase it like this in the doc then.

"Currently, the **Google Drive connector** supports two types of templates: create a folder and create a file from a template."

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 4 above)
Comment (christinaausley) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:29]: So do you mean...
"In the **Operation Details** section, you can set the **Additional properties or metadata** field to Google Drive compatible properties. For example, *description* of the folder."

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 1 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Done

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 3 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Done

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 4 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Done

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 5 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Done

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 6 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Done

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 7 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: done

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 8 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:58]: Yes, correct. Updated.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 9 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Rewrote the whole section

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 10 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Updated

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 3 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Updated

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 4 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Yes, updated.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 5 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Done

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 9 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Rewrote the whole section

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 11 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Done

**Comment by igpetrov** [comment]
Notes on the latest changes:
- Addressed comments
- Added documentation related to the OAuth refresh token authorization
- Updated screenshot with updated template

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 2 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Added Google Drive link. For templates we don't have explicit reference, except one mentioned in FAQ section. Let me know if you think, we should add that one.

**Comment by igpetrov** [comment]
Screenshot 2022-08-30 at 13 24 19
Screenshot 2022-08-30 at 13 25 22
Screenshot 2022-08-30 at 13 25 48
Screenshot 2022-08-30 at 13 26 08
Screenshot 2022-08-30 at 13 26 29
Screenshot 2022-08-30 at 13 26 43
Screenshot 2022-08-30 at 13 26 58
Screenshot 2022-08-30 at 13 27 08

**Review by christinaausley — APPROVED** [review]
Updated my branch and committed some technical adjustments as it looks like I didn't push last time. Looks good on my end!

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md [hunk 12]
```diff
@@ -9,5 +9,6 @@ This section gives an overview of the **out-of-the-box connectors** available in
 - [REST connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
 - [SendGrid connector](sendgrid.md) - Quickly send emails from your BPMN processes.
 - [Slack connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
+- [Google Drive connector](googledrive.md) - Create empty folders or documents on [Google Drive](https://www.google.com/drive/) from templates from your BPMN process.
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:12]: I propose to order them alphabetically

File: sidebars.js [hunk 13]
```diff
@@ -310,6 +310,7 @@ module.exports = {
                 "components/integration-framework/connectors/out-of-the-box-connectors/rest",
                 "components/integration-framework/connectors/out-of-the-box-connectors/sendgrid",
                 "components/integration-framework/connectors/out-of-the-box-connectors/slack",
+                "components/integration-framework/connectors/out-of-the-box-connectors/googledrive",
```
Comment (MaxTru) [inline sidebars.js:313]: I propose to order them alphabetically

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+
+To start working with the **Google Drive Connector**, a relevant OAuth token must be configured and stored as a secret in the user's cluster. The token must have permission to read and create a folder and/or files from a desired Google Drive instance. Follow the steps from the [appendix](#appendix--faq) to find out more about creating an OAuth token and giving relevant permissions.
+
+## Create a Google Drive Connector task
+
+Currently, the **Google Drive Connector** supports two types of templates: create a folder and create a file from a template.
+
+To use a **Google Drive Connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu icon or create a new Connector task by using the **Append Connector** context menu. Follow our [guide on using Connectors](../use-connectors.md) to learn more.
+
+## Make your Google Drive Connector for creating a folder executable
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:19]: To remove redundancy and get away from this long heading, have you considered a structure like this:

```
## Make your Google Drive Connector executable

To make the **Google Drive Connector** executable, fill out the mandatory fields highlighted in red in the properties panel:

![Google Drive Connector new folder example](../img/connectors-googledrive-new-folder-filled.png)

For all oeprations, you need to fill the required credentials in the **Authentication** section. See the [relevant appendix entry](#how-can-i-authenticate-my-connector) to find out more.

You can then decide between two operations: [creating a folder](link) or [creating a file from template](link)

### Create a new folder
...

### Create a file from a template
...
```

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+See [related article](https://developers.google.com/identity/protocols/oauth2/web-server).
+
+:::warning
+The following code snippets are for demonstration purposes only and must be not used for real production systems due to security concerns.
+:::
+
+Assuming you have created an OAuth client, you should an OAuth Client ID and an OAuth Client secret. Run the following code snippet that prints the refresh token in terminal:
+
+```python
+TBD - prototype code will be added soon
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:119]: Note this is open

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+
+### Where do I get a parent folder ID?
+
+It is in the URL.
+
+![Google Drive Connector get parent folder ID](../img/connectors-googledrive-get-parent-folder-id.png)
+
+### How do I set additional properties or metadata?
+
+You can set any property from the Google Drive Create API. For example:
```
Comment (MaxTru) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:130]: Can we link to Google Drive Create API explaining which properties exist?

**Review by MaxTru — CHANGES_REQUESTED** [review]
Please see some small comments

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md (hunk 12 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md]: Done

File: sidebars.js (hunk 13 above)
Comment (igpetrov) [reply sidebars.js]: Done

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 14 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Agreed. Updated.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 15 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md]: Added

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md [hunk 17]
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
 
 This section gives an overview of the **out-of-the-box Connectors** available in Camunda Platform 8:
 
-- [REST Connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
-- [SendGrid Connector](sendgrid.md) - Quickly send emails from your BPMN processes.
-- [Slack Connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
+- [Google Drive connector](googledrive.md) - Create empty folders or documents on [Google Drive](https://www.google.com/drive/) from templates from your BPMN process.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:9]: Update text to: 
"Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process."

Comment (PreetiNirwal) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:9]: Also can the order of the list be: 
1. REST Connector
2. SendGrid Connector
3. Slack Connector
4. Google Drive Connector
(For our other new connectors coming)
5. AWS Simple Queue Service Connector
6. AWS Lambda Connector

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 18]
```diff
@@ -0,0 +1,245 @@
+---
+id: googledrive
+title: Google Drive Connector
+description: Create empty folders or documents from templates from your BPMN process.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:4]: Pls update text to:
"Create folders or files from a Google Drive template from your BPMN process."

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+---
+
+The **Google Drive Connector** allows you to create empty folders or documents on [Google Drive](https://drive.google.com/) from templates from your BPMN process.
+
+## Prerequisites
+
+To start working with the **Google Drive Connector**, a relevant OAuth token must be configured and stored as a Secret in the user's cluster. The token must have permission to read and create a folder and/or files from a desired Google Drive instance. Follow the steps from the [appendix](#appendix--faq) to find out more about creating an OAuth token and giving relevant permissions.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:11]: Pls update text from: "....stored as a Secret in the user's cluster." 
to:  "....stored as a Secret in your cluster."

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+
+## Create a Google Drive Connector task
+
+Currently, the **Google Drive Connector** supports two types of templates: create a folder and create a file from a template.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:15]: Pls update text to:
"Currently, the **Google Drive Connector** supports two types of operations: create a folder and create a file from a template."

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 21]
```diff
@@ -0,0 +1,245 @@
...
+
+![Google Drive Connector new file from template example](../img/connectors-googledrive-new-file-filled.png)
+
+1. Set required credentials in the **Authentication** section. See the [relevant appendix entry](#how-can-i-authenticate-my-connector) to find out more.
+2. In the **Select Operation** section, set the field value **Operation Type** as **Create File from template**.
+3. In the **Operation Details** section, set the field **File name** as the desired name of a folder you wish to create. You can use FEEL expressions here.
+4. In the **Operation Details** section, set the field **Template ID** of the desired template.
+5. _(optional)_ In the **Operation Details** section, set the field **Parent folder ID** to the desired parent, inside which a new file will be created. Keep in mind that if not specified, a new folder will be created in the Google Drive root folder of a user who owns the OAuth token.
+6. In the **Operation Details** section, set the field **Template variables** as desired variables that will be applied to the template. The template variables a compatible with the Google Docs Requests API. This property requires FEEL input.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:42]: Pls update text from: "The template variables a compatible with the Google Docs Requests API."
to: "The template variables are compatible with the Google Docs Requests API."

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 22]
```diff
@@ -0,0 +1,245 @@
...
+  "myNewReportFolderId": response.googleDriveResourceId,
+  "myNewReportFolderUrl": response.googleDriveResourceUrl
+}
+```
+
+## Appendix & FAQ
+
+### What Google API does the Google Drive Connector use to create a folder?
+
+The **Google Drive Connector** uses Google Drive [`Files:Create`](https://developers.google.com/drive/api/v3/reference/files/create) API endpoint.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:69]: Pls update text to: "The **Google Drive Connector** uses the Google Drive...."

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 23]
```diff
@@ -0,0 +1,245 @@
...
+
+### What Google API does the Google Drive Connector use to create a file from template?
+
+The **Google Drive Connector** uses Google Drive [`Files:Copy`](https://developers.google.com/drive/api/v3/reference/files/copy) API endpoint to copy an original template. Afterwards, the **Google Drive Connector** utilizes Google Docs [Merge](https://developers.google.com/docs/api/how-tos/merge) approach via [`Documents:BatchUpdate`](https://developers.google.com/docs/api/reference/rest/v1/documents/batchUpdate) Google Docs API method.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:73]: Pls update first sentence to: "The Google Drive Connector uses the Google Drive...."

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md (hunk 17 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:9]: Addressed in a new revision. Will publish shortly.

Regarding connectors order - there was a suggestion from Max to keep them alphabetical.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 18 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:4]: Addressed in a new revision. Will publish shortly.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 19 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:11]: Addressed in a new revision. Will publish shortly.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 20 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:15]: Addressed in a new revision. Will publish shortly.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 21 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:42]: Addressed in a new revision. Will publish shortly.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 22 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:69]: Addressed in a new revision. Will publish shortly.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md (hunk 23 above)
Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:73]: Addressed in a new revision. Will publish shortly.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md [hunk 24]
```diff
@@ -0,0 +1,245 @@
...
+
+### Where do I get Template ID?
+
+It is in the URL.
+
+![Google Drive Connector get template ID](../img/connectors-googledrive-get-template-id.png)
+
+### Can you show me an example of a valid template?
+
+The valid template looks like the one on a screenshot.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:163]: Pls update text to: "A valid template looks like the following screenshot."

Comment (igpetrov) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:163]: Addressed in a new revision. Will publish shortly.

**Comment by igpetrov** [comment]
Published new new revision

**Comment by igpetrov** [comment]
@PreetiNirwal, updated wording and screenshots.

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md (hunk 17 above)
Comment (PreetiNirwal) [reply docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md]: Ok cool (on the alphabetical) piece!

**Review by PreetiNirwal — DISMISSED** [review]
Thanks for the updates! 
Looks great!

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md [hunk 25]
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
 
 This section gives an overview of the **out-of-the-box Connectors** available in Camunda Platform 8:
 
-- [REST Connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
-- [SendGrid Connector](sendgrid.md) - Quickly send emails from your BPMN processes.
-- [Slack Connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
+- [Google Drive connector](googledrive.md) - Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:9]:
```suggestion
- [Google Drive Connector](googledrive.md) - Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process.
```

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md [hunk 26]
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
...
+- [REST connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:10]:
```suggestion
- [REST Connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
```

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md [hunk 27]
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
...
+- [SendGrid connector](sendgrid.md) - Quickly send emails from your BPMN processes.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:11]:
```suggestion
- [SendGrid Connector](sendgrid.md) - Quickly send emails from your BPMN processes.
```

File: docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md [hunk 28]
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
...
+- [Slack connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
```
Comment (PreetiNirwal) [inline docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:12]:
```suggestion
- [Slack Connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
```

**Review by PreetiNirwal — APPROVED** [review]
Minor update only, we are capitalising Connectors.

**Review by christinaausley — APPROVED** [review]
Technical adjustments in latest commit.

**Review by christinaausley — APPROVED** [review]

**Pull request merged**